src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -108,10 +102,8 @@ private CompilationWithAnalyzers(Compilation compilation, ImmutableArray<Diagnos
             _analysisOptions = analysisOptions;
             _cancellationToken = cancellationToken;
 
-            _compilationData = new CompilationData(_compilation);
             _analysisResultBuilder = new AnalysisResultBuilder(analysisOptions.LogAnalyzerExecutionTime, analyzers, _analysisOptions.Options?.AdditionalFiles ?? ImmutableArray<AdditionalText>.Empty);
-            _analyzerManager = new AnalyzerManager(analyzers);
-            _driverPool = new ObjectPool<AnalyzerDriver>(() => _compilation.CreateAnalyzerDriver(analyzers, _analyzerManager, severityFilter: SeverityFilter.None));
+            _compilationAnalysisScope = new AnalysisScope(_compilation, _analysisOptions.Options, _analyzers, hasAllAnalyzers: true, _analysisOptions.ConcurrentAnalysis, categorizeDiagnostics: true);
```

Comment:
what does 'hasAllAnalyzers' mean here, and `categorizeDiagnostics`?  Effectively, i'm asking why the true values are appropriate here.  Consider doc'ing.

Reply:
Sure, will add comments

Reply:
I refactored and cleaned up the code in `AnalysisScope` so this perf-related flag is now computed with `AnalysisScope` type itself, callers don't need to pass it in.